staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -495,6 +497,20 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	if a.group.MetaGroupVersion != nil {
 		reqScope.MetaGroupVersion = *a.group.MetaGroupVersion
 	}
+	if a.group.OpenAPIConfig != nil {
+		openAPIDefinitions, err := openapibuilder.BuildOpenAPIDefinitionsForResource(defaultVersionedObject, a.group.OpenAPIConfig)
+		if err != nil {
+			return nil, fmt.Errorf("unable to build openapi definitions for %v: %v", fqKindToRegister, err)
+		}
+		reqScope.OpenAPISchema, err = utilopenapi.ToProtoSchema(openAPIDefinitions, fqKindToRegister)
+		// It is necessary to ignore IsNotRegisteredErrors because currently the openapi definition for
+		// apiextensions.v1beta1.CustomResourceDefinition does not have the property
+		// "x-kubernetes-group-version-kind" set correctly
+		// TODO(jennybuckley): Fix apiextensions-apiserver to correctly set the "x-kubernetes-group-version-kind" property
+		if err != nil && !runtime.IsNotRegisteredError(err) {
```

Comment:
This still isn't scoped tightly enough. Ignoring any "not registered" error would let through new types/servers we introduced that didn't set up openapi correctly from the beginning.

Reply:
@liggitt
Oh, I see so do we want to explicitly check if the gvk is a CRD?

FWIW, an IsNotRegisteredError will only be returned by ToProtoSchema if the definitions/models are able to be built and it gets to the last step of looking up the schema by the x-kubernetes-group-version-kind tag and can't find it.

Reply:
It looks like kube apiserver is constructing openapi passing in a scheme that doesn't have all the extension types included in it: 

https://github.com/kubernetes/kubernetes/blob/6187898b60954c71807fc44897f4b632e5d263e8/cmd/kube-apiserver/app/server.go

If we're going to keep kube-apiserver constructing a monolithic openapi doc at once, the DefaultOpenAPIConfig method should be updated to not assume a single scheme (kube-apiserver and extensions-apiserver each have their own scheme for their types).

Reply:
Yes, I was hoping for a 1-item whitelist of GVKs that are allowed to fail :)

Reply:
fwiw, the last commit at https://github.com/liggitt/kubernetes/commits/jennybuckley-expose-openapi-to-registry3 corrects the kube-apiserver openapi definition namer to process extension and aggregator types correctly